velox/exec/fuzzer/AggregationFuzzerBase.cpp

```diff
@@ -353,18 +348,63 @@ std::vector<RowVectorPtr> AggregationFuzzerBase::generateInputDataWithRowNumber(
           generator->generate(signature.args, vectorFuzzer_, rng_, pool_.get());
     }
 
-    // Number of partitions is randomly generated and is at least 1.
-    auto numPartitions = size ? randInt(1, size) : 1;
-    auto indices = vectorFuzzer_.fuzzIndices(size, numPartitions);
-    auto nulls = vectorFuzzer_.fuzzNulls(size);
+    // Number of peer groups is randomly generated and is at least 1. Each input
```

Comment:
Comments should explain the idea behind the algorithm and not the steps of code. Those are evident if the code is readable.

Would suggest rephrase as 

Some window functions like 'rank' have semantics influenced by "peer" rows. Peer rows are rows in the partition with the same order by key. In rank (and dense_rank) function peer rows have the same function result value. The goal of this code is to influence the fuzzer to generate such data.

To build such rows the code separates the notions of "peer" groups and "partition" groups  during data generation. A number of peers are chosen between (1, size) of the input. Rows with the same peer number have the same order by keys. This means that there are sets of rows in the input data which will have the same order by key. 

Each peer is then mapped to a partition group. Rows in the same partition group have the same partition keys. So this would mean each partition has multiple rows with the same order by key and also there will be partitions with multiple sets of rows with the same order by key.

This style of data generation is preferable for window functions.

This can be followed with your example.

Reply:
Thanks for the feedback, updated accordingly.